lib/postgrex/extensions/binary.ex

```diff
@@ -304,6 +306,30 @@ defmodule Postgrex.Extensions.Binary do
     [flags|bin]
   end
 
+  defp encode_hstore(bin) do
+    keys_and_values = Enum.reduce bin, "", fn ({key, value}, acc) ->
+        [acc, encode_hstore_key(key), encode_hstore_value(value)]
+    end
+    :erlang.iolist_to_binary([<<Map.size(bin)::int32>> | keys_and_values])
+  end
+
+  defp encode_hstore_key(key) when not is_nil(key) do
+    encode_hstore_value key
+  end
+
+  defp encode_hstore_key(_key)  do
+    raise ArgumentError, message: "Hstore keys cannot be nil!"
```

Comment:
I would actually vote for keeping it. It is going to be much clearer and I assume it will be a common mistake.